File: Client/go/list/list_request_builder.go
package list

import (
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
)

// ListRequestBuilder builds and executes requests for operations under \list
type ListRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// Chromosomes the chromosomes property
// returns a *ChromosomesRequestBuilder when successful
func (m *ListRequestBuilder) Chromosomes()(*ChromosomesRequestBuilder) {
    return NewChromosomesRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// NewListRequestBuilderInternal instantiates a new ListRequestBuilder and sets the default values.
func NewListRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ListRequestBuilder) {
    m := &ListRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/list", pathParameters),
    }
    return m
}
// NewListRequestBuilder instantiates a new ListRequestBuilder and sets the default values.
func NewListRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ListRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewListRequestBuilderInternal(urlParams, requestAdapter)
}
// Files the files property
// returns a *FilesRequestBuilder when successful
func (m *ListRequestBuilder) Files()(*FilesRequestBuilder) {
    return NewFilesRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// HubGenomes the hubGenomes property
// returns a *HubGenomesRequestBuilder when successful
func (m *ListRequestBuilder) HubGenomes()(*HubGenomesRequestBuilder) {
    return NewHubGenomesRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// PublicHubs the publicHubs property
// returns a *PublicHubsRequestBuilder when successful
func (m *ListRequestBuilder) PublicHubs()(*PublicHubsRequestBuilder) {
    return NewPublicHubsRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// Schema the schema property
// returns a *SchemaRequestBuilder when successful
func (m *ListRequestBuilder) Schema()(*SchemaRequestBuilder) {
    return NewSchemaRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// Tracks the tracks property
// returns a *TracksRequestBuilder when successful
func (m *ListRequestBuilder) Tracks()(*TracksRequestBuilder) {
    return NewTracksRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// UcscGenomes the ucscGenomes property
// returns a *UcscGenomesRequestBuilder when successful
func (m *ListRequestBuilder) UcscGenomes()(*UcscGenomesRequestBuilder) {
    return NewUcscGenomesRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}

File: Client/ruby/list/list_request_builder.rb
require 'microsoft_kiota_abstractions'
require_relative '../api_sdk'
require_relative './chromosomes/chromosomes_request_builder'
require_relative './files/files_request_builder'
require_relative './hub_genomes/hub_genomes_request_builder'
require_relative './list'
require_relative './public_hubs/public_hubs_request_builder'
require_relative './schema/schema_request_builder'
require_relative './tracks/tracks_request_builder'
require_relative './ucsc_genomes/ucsc_genomes_request_builder'

module ApiSdk
    module List
        ## 
        # Builds and executes requests for operations under \list
        class ListRequestBuilder < MicrosoftKiotaAbstractions::BaseRequestBuilder
            
            ## 
            # The chromosomes property
            def chromosomes()
                return ApiSdk::List::Chromosomes::ChromosomesRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The files property
            def files()
                return ApiSdk::List::Files::FilesRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The hubGenomes property
            def hub_genomes()
                return ApiSdk::List::HubGenomes::HubGenomesRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The publicHubs property
            def public_hubs()
                return ApiSdk::List::PublicHubs::PublicHubsRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The schema property
            def schema()
                return ApiSdk::List::Schema::SchemaRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The tracks property
            def tracks()
                return ApiSdk::List::Tracks::TracksRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            # The ucscGenomes property
            def ucsc_genomes()
                return ApiSdk::List::UcscGenomes::UcscGenomesRequestBuilder.new(@path_parameters, @request_adapter)
            end
            ## 
            ## Instantiates a new ListRequestBuilder and sets the default values.
            ## @param path_parameters Path parameters for the request
            ## @param request_adapter The request adapter to use to execute the requests.
            ## @return a void
            ## 
            def initialize(path_parameters, request_adapter)
                super(path_parameters, request_adapter, "{+baseurl}/list")
            end
        end
    end
end
